CRM/Pledge/BAO/Pledge.php

```diff
@@ -1204,15 +1204,17 @@ public static function buildRecurParams($params) {
   public static function getPledgeStartDate($date, $pledgeBlock) {
     $startDate = (array) json_decode($pledgeBlock['pledge_start_date']);
     list($field, $value) = each($startDate);
-    if (!CRM_Utils_Array::value('is_pledge_start_date_visible', $pledgeBlock)) {
-      return date('Ymd', strtotime($value));
-    }
-    if (!CRM_Utils_Array::value('is_pledge_start_date_editable', $pledgeBlock)) {
+    if (!empty($date) && !CRM_Utils_Array::value('is_pledge_start_date_editable', $pledgeBlock)) {
       return $date;
     }
+    if (empty($date)) {
+      $date = $value;
+    }
     switch ($field) {
       case 'contribution_date':
-        $date = date('Ymd');
+        if (empty($date)) {
+          $date = date('Ymd');
```

Comment:
Again, how does this work with localization of date formats, and why Ymd here but m/d/Y above?